src/fsharp/FSharp.Core/seq.fs

```diff
@@ -1577,57 +1577,56 @@ namespace Microsoft.FSharp.Collections
 
         [<CompiledName("Sum")>]
         let inline sum (source: seq< (^a) >) : ^a = 
-            use e = source.GetEnumerator() 
             let mutable acc = LanguagePrimitives.GenericZero< (^a) >
-            while e.MoveNext() do
-                acc <- Checked.(+) acc e.Current
+            for item in source do
+                acc <- Checked.(+) acc item
```

Comment:
@dsyme I have reverted the seq.fs changes. I have watched your recent video - thank for the explanation why this is not as straightforward as I had hoped for (and for kindly mentioning Spreads). I thought that just by providing my enumerators for Series as structs both C# and F# compilers will do the rest just by magic.